src/api-engine/api/routes/node/views.py

```diff
@@ -555,15 +555,19 @@ def destroy(self, request, pk=None):
                 node = Node.objects.get(id=pk)
                 infos = self._agent_params(pk)
                 agent = AgentHandler(infos)
+                agent_exist = agent.get()
                 node.status = "removing"
                 node.save()
                 if node.type == "orderer":
                     orderer_cnt = Node.objects.filter(
                         type="orderer", organization__network=node.organization.network).count()
                     if orderer_cnt == 1:
                         raise ResourceInUse
-                agent.stop()
-                res = True if agent.delete() else False
+                if agent_exist:
```

Comment:
I'd like to check the node status instead, if a node is created but does not have a docker container, the status should be `created` (need to double check this).

Reply:
Nice to have, we could sync the status in database when api-engine in the startup process.

Reply:
If  1) agent does not exist;  OR 2) agent exists, but container record does not exists in DB. 
We should return OK.
Else:
Try to issue delete operation.

Reply:
Try several times, if still with exception, then return false.
TODO: need optimization in future release.

Reply:
@yeasy @zhuyuanmao 

Could you indicate where to find the container record in DB? 

I didn't find an appropriate column in the Node table(and other tables such as Agent) to query the container record.

Reply:
@xichen1  There is a field that records the id of the container in the Node model, if one container was initialized for the node, there should be an ID that relates to the docker container. You'll be able to take this advantage to find the corresponding container record. 

https://sourcegraph.com/github.com/hyperledger/cello/-/blob/src/api-engine/api/models.py?L540-544&subtree=true

Reply:
Didn't notice this id, thank you!